test/OpenTelemetry.Exporter.Zipkin.Tests/ZipkinExporterTests.cs

```diff
@@ -89,6 +90,52 @@ public void ZipkinExporter_BadArgs()
             Assert.Throws<ArgumentNullException>(() => builder.AddZipkinExporter());
         }
 
+        [Fact]
+        public void ZipkinExporter_SuppresssesInstrumentation()
+        {
+            const string ActivitySourceName = "zipkin.test";
+            Guid requestId = Guid.NewGuid();
+            TestActivityProcessor testActivityProcessor = new TestActivityProcessor();
+
+            int endCalledCount = 0;
+
+            testActivityProcessor.EndAction =
+                (a) =>
+                {
+                    endCalledCount++;
+                };
+
+            var exporterOptions = new ZipkinExporterOptions();
+            exporterOptions.ServiceName = "test-zipkin";
+            exporterOptions.Endpoint = new Uri($"http://{this.testServerHost}:{this.testServerPort}/api/v2/spans?requestId={requestId}");
+            var zipkinExporter = new ZipkinExporter(exporterOptions);
+            var exportActivityProcessor = new BatchExportActivityProcessor(zipkinExporter, scheduledDelayMilliseconds: 1);
+
+            var openTelemetrySdk = Sdk.CreateTracerProviderBuilder()
+                .AddSource(ActivitySourceName)
+                .AddProcessor(testActivityProcessor)
+                .AddProcessor(exportActivityProcessor)
+                .AddHttpClientInstrumentation()
+                .Build();
+
+            var source = new ActivitySource(ActivitySourceName);
+            var activity = source.StartActivity("Test Zipkin Activity");
+            activity?.Stop();
+
+            // This test currently uses the BatchExportActivityProcessor, so we
+            // wait a short amount of time for the exporter to be invoked.
+            // Ideally, once the Zipkin exporter is able to be configured with
+            // the SimpleExportActivityProcessor the test could use that and
+            // this wait would not be needed.
+            // Simply using the SimpleExportActivityProcessor in this test does
+            // not work at the moment because doing so causes deadlock in the
+            // exporter as it uses .GetAwaiter().GetResult() to make the HTTP
+            // call synchronously.
```

Comment:
This is probably the more interesting part of this PR. There is a TODO in `AddZipkinExporter` to allow using the `SimpleExportActivityProcessor`, but using it causes a deadlock in the test due to the way that Zipkin makes its HTTP call synchronously.

Reply:
I guess `SimpleExportActivityProcessor` should work, right?
Someone calls HTTP client -> Activity created -> SimpleExportActivityProcessor called -> Zipkin exporter called -> SuppressInstrumentation flag is set -> HTTP client called to export data -> Activity created -> legacy activity adapter called -> check the SuppressInstrumentation flag and drop data on the floor.

Reply:
Yes, it _should_ work but this line deadlocks:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Exporter.Zipkin/ZipkinExporter.cs#L71

This line is a hack. It may not deadlock in all circumstances, but it does in the context of the test runner.

General guidance for async calls is to not block on async code https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html.

Reply:
Probably something like this:

```csharp
Task.Run(async () => 
{
    using var scope = SuppressInstrumentationScope.Begin();
    await httpClient.PostAsync(...);
}).ConfigureAwait(false).GetAwaiter().GetResult();
```

Reply:
Yep, wrapping in `Task.Run` was my thought as well, though haven't tried it out yet to make sure. Mind if I follow this up in a separate PR?

Reply:
Sounds good.